dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/math/BeamSqlMathUnaryExpression.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.dsls.sql.interpreter.operator.math;
+
+import java.util.List;
+
+import org.apache.beam.dsls.sql.interpreter.operator.BeamSqlExpression;
+import org.apache.beam.dsls.sql.interpreter.operator.BeamSqlPrimitive;
+import org.apache.beam.dsls.sql.schema.BeamSQLRow;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+/**
+ * Base class for all unary functions such as
+ * ABS, SQRT, LN, LOG10, EXP, CEIL, FLOOR, RAND, ACOS,
+ * ASIN, ATAN, COS, COT, DEGREES, RADIANS, SIGN, SIN, TAN.
+ */
+public abstract class BeamSqlMathUnaryExpression extends BeamSqlExpression {
+
+  public BeamSqlMathUnaryExpression(List<BeamSqlExpression> operands) {
+    super(operands, SqlTypeName.ANY);
+  }
+
+  @Override public boolean accept() {
+    boolean acceptance = false;
+
+    if (numberOfOperands() == 1 && SqlTypeName.NUMERIC_TYPES.contains(opType(0))) {
+      acceptance = true;
+    }
+    return acceptance;
+  }
+
+  @Override public BeamSqlPrimitive<? extends Number> evaluate(BeamSQLRow inputRecord) {
+    BeamSqlExpression operand = op(0);
+    if (SqlTypeName.INT_TYPES.contains(operand.getOutputType())) {
```

Comment:
@xumingming  Another implementation design :

public abstract class BeamSqlMathUnaryExpression extends BeamSqlExpression {

  public BeamSqlMathUnaryExpression(List<BeamSqlExpression> operands) {
    super(operands, SqlTypeName.ANY);
  }

  @Override public boolean accept() {
    boolean acceptance = false;

    if (numberOfOperands() == 1 && SqlTypeName.NUMERIC_TYPES.contains(opType(0))) {
      acceptance = true;
    }
    return acceptance;
  }

  @Override public BeamSqlPrimitive<? extends Number> evaluate(BeamSQLRow inputRecord) {
    BeamSqlExpression operand = op(0);
    TypeValueTuple typeValueOp = getTypeValueTuple(operand.evaluate(inputRecord).getValue());
    TypeValueTuple typeValueResult = getTypeValueTuple(
        calculate(typeValueOp.getType(), typeValueOp.getValue()));
    return BeamSqlPrimitive.of(typeValueResult.getType(), typeValueResult.getValue());
  }

  /**
   * Method for generating tuple of ( {@link SqlTypeName}, {@link Number} ).
   * @param value
   * @return {@link TypeValueTuple}
   */
  private TypeValueTuple getTypeValueTuple(Object value) {
    if (value instanceof Long) {
      return TypeValueTuple.of(SqlTypeName.BIGINT, Long.valueOf(value.toString()));
    } else if (value instanceof Integer) {
      return TypeValueTuple.of(SqlTypeName.INTEGER, Integer.valueOf(value.toString()));
    } else if (value instanceof Short) {
      return TypeValueTuple.of(SqlTypeName.SMALLINT, Short.valueOf(value.toString()));
    } else if (value instanceof Float) {
      return TypeValueTuple.of(SqlTypeName.FLOAT, Float.valueOf(value.toString()));
    } else if (value instanceof Byte) {
      return TypeValueTuple.of(SqlTypeName.TINYINT, Byte.valueOf(value.toString()));
    } else {
      return TypeValueTuple.of(SqlTypeName.DOUBLE, Double.valueOf(value.toString()));
    }
  }

  /**
   * For the operands of other type {@link SqlTypeName#NUMERIC_TYPES}.
   * */

  public abstract Number calculate(SqlTypeName type, Number op);

  /**
   * Tuple class for storing {@link SqlTypeName#NUMERIC_TYPES} and its numeric values.
   */
  private static class TypeValueTuple {

    private SqlTypeName type;
    private Number value;

    private TypeValueTuple(SqlTypeName type, Number value) {
      this.type = type;
      this.value = value;
    }

    public static TypeValueTuple of(SqlTypeName type, Number value) {
      return new TypeValueTuple(type, value);
    }

    public SqlTypeName getType() {
      return type;
    }

    public Number getValue() {
      return value;
    }
  }
}